scripts/o3de/o3de/repo_properties.py

```diff
@@ -0,0 +1,399 @@
+
+logger = logging.getLogger('o3de.repo_properties')
+logging.basicConfig(format=utils.LOG_FORMAT)
+
+
+def merge_json_data(json_path: pathlib.Path, json_data: dict) -> dict:
+    """
+    Merges the json data with the existing json data if it exists.
+    If the json data does not exist, it is created.
+    The json data is saved to the json path.
+    Returns the merged json data.
+    
+    Args:
+        json_path (pathlib.Path): The path to the json file.
+        json_data (dict): The json data to merge with the existing json data.
+    
+    Returns:
+        dict: The merged json data.
+    
+    Raises:
+        FileNotFoundError: If the json path does not exist.
+        ValueError: If the json data is not a dict.
+    """
+    merged_json_data = {}
+    if json_path.exists():
+        existing_json_data = {}
+        with open(json_path, 'r') as f:
+            existing_json_data = json.load(f)
+
+        merged_json_data = existing_json_data | json_data
+
+        if not 'version' in merged_json_data:
+            merged_json_data['version'] = '1.0.0'
+
+        with open(json_path, 'w') as f:
+            f.write(json.dumps(merged_json_data, indent=4) + '\n')
+
+    return merged_json_data
+
+def create_release(src_data_path: pathlib.Path, 
+                   json_data_path: pathlib.Path, 
+                   archive_filename: pathlib.PurePath, 
+                   releases_path: pathlib.Path, 
+                   repo_uri:str,
+                   download_prefix:str) -> dict:
+    """
+    Creates a release for the given src_data_path.
+    The release is saved to the releases_path.
+    Returns the json data.
+    
+    Args:
+        src_data_path (pathlib.Path): The path to the src root folder.
+        json_data_path (pathlib.Path): The path to the json data.
+        name (str): The object name.
+        releases_path (pathlib.Path): The path where the release archive should be output.
+        repo_uri (str): The uri of the repo.
+        download_prefix (str): The prefix of the download uri.
+    
+    Returns:
+        dict: The json data.
+    
+    Raises:
+        FileNotFoundError: If the src_data_path does not exist.
+        FileNotFoundError: If the json_data_path does not exist.
+        ValueError: If the json_data_path is not a dict.
+    """
+    # basename = f'{object_name}_release_1.0.0'
+    # zip_filename = f'{archive_filename}.zip'
+    basename = archive_filename.stem
+    zip_filename = archive_filename
+
+    json_data = merge_json_data(json_data_path, {
+        'repo_uri':repo_uri,
+        'download_source_uri':f'{download_prefix}/{zip_filename}',
+    })
+
+    logging.info(f"Creating '{releases_path / zip_filename}'")
+
+    # create the release zip file
+    shutil.make_archive(releases_path / basename, 'zip', src_data_path)
+
+    zip_path = releases_path / zip_filename
+
+    with zip_path.open('rb') as f:
+        json_data['sha256'] = hashlib.sha256(f.read()).hexdigest()
```

Comment:
What if the user doesn't want to use a `zip`, but another commonly used format for different platform such as Linux or MacOS?
On Linux `tar.gz` and `tar.xz` is popular and MacOS `.xz` is popular

How about data driving the archive format via an argparse argument?

That way the user can choose the options that [shutil.make_archive](https://docs.python.org/3/library/shutil.html#shutil.make_archive) makes available.

Reply:
Currently [our download archive code](https://github.com/o3de/o3de/blob/d9285c8e0c7804b64f51762ac6ef3efbd4907e83/scripts/o3de/o3de/download.py#L168) is looking for zip archive.  But we should definitely make this more flexible in the future!